sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageChannelMode.java

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common;
+
+/**
+ * Mode to open a Storage-backed SeekableByteChannel.
+ */
+public enum StorageChannelMode {
+    /**
+     * Open for read-only mode.
+     */
+    READ,
+
+    /**
+     * Open for write-only mode.
+     */
+    // TODO (jaschrep): should this be `APPEND`?
```

Comment:
I feel that `WRITE` is the more generalizable solution here, for example you could upload to a middle part of a file which isn't really an append